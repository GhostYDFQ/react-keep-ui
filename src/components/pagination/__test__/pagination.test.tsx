import React from "react";
import { render, cleanup } from "@testing-library/react";
import { fireEvent } from "@testing-library/dom";
import { Pagination } from "../index";
import { color, typography } from "../../shared/styles";

describe("test Pagination component", () => {
    it('render correctly', () => {
        const wrapper = render(
            <Pagination />
        );
        expect(wrapper).toMatchSnapshot();
    });
})
