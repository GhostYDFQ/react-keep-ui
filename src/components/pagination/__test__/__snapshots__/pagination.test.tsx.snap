// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Pagination component render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-jSFkmK dWlMqp"
      >
        <li>
          <button
            class="sc-eCApGN jphxbT"
          >
            <span
              class="sc-dlnjPT gbwVbp"
            >
              <svg
                class="sc-bdnylx fIWMCQ"
                color="#FF4785"
                height="20px"
                viewBox="0 0 1024 1024"
                width="20px"
              >
                <path
                  class="sc-gtssRu cdMHpp"
                  color="#FF4785"
                  d="M257.93 511.976c0-10.236 3.902-20.47 11.71-28.282l344.098-344.158c15.622-15.624 40.946-15.624 56.57-0.006 15.622 15.622 15.624 40.948 0.004 56.568l-315.82 315.876 315.868 315.922c15.618 15.624 15.618 40.952-0.004 56.568-15.622 15.62-40.95 15.618-56.57-0.006l-344.146-344.202c-7.808-7.81-11.71-18.044-11.71-28.28z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-eCApGN jphxbT"
          >
            <span
              class="sc-dlnjPT gbwVbp"
            >
              9
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-eCApGN jphxbT"
          >
            <span
              class="sc-dlnjPT gbwVbp"
            >
              10
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-eCApGN kBsjdL"
          >
            <span
              class="sc-dlnjPT gbwVbp"
            >
              11
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-eCApGN jphxbT"
          >
            <span
              class="sc-dlnjPT gbwVbp"
            >
              12
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-eCApGN jphxbT"
          >
            <span
              class="sc-dlnjPT gbwVbp"
            >
              13
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-eCApGN jphxbT"
          >
            <span
              class="sc-dlnjPT gbwVbp"
            >
              <svg
                class="sc-bdnylx fIWMCQ"
                color="#FF4785"
                height="20px"
                viewBox="0 0 1024 1024"
                width="20px"
              >
                <path
                  class="sc-gtssRu cdMHpp"
                  color="#FF4785"
                  d="M768.072 514.022c0 10.236-3.904 20.47-11.712 28.282l-344.098 344.156c-15.62 15.624-40.946 15.624-56.568 0.006-15.622-15.622-15.624-40.948-0.006-56.568l315.82-315.876-315.868-315.922c-15.618-15.624-15.618-40.952 0.004-56.568 15.624-15.62 40.95-15.618 56.57 0.006l344.144 344.204c7.81 7.81 11.714 18.044 11.714 28.28z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-jSFkmK dWlMqp"
    >
      <li>
        <button
          class="sc-eCApGN jphxbT"
        >
          <span
            class="sc-dlnjPT gbwVbp"
          >
            <svg
              class="sc-bdnylx fIWMCQ"
              color="#FF4785"
              height="20px"
              viewBox="0 0 1024 1024"
              width="20px"
            >
              <path
                class="sc-gtssRu cdMHpp"
                color="#FF4785"
                d="M257.93 511.976c0-10.236 3.902-20.47 11.71-28.282l344.098-344.158c15.622-15.624 40.946-15.624 56.57-0.006 15.622 15.622 15.624 40.948 0.004 56.568l-315.82 315.876 315.868 315.922c15.618 15.624 15.618 40.952-0.004 56.568-15.622 15.62-40.95 15.618-56.57-0.006l-344.146-344.202c-7.808-7.81-11.71-18.044-11.71-28.28z"
              />
            </svg>
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-eCApGN jphxbT"
        >
          <span
            class="sc-dlnjPT gbwVbp"
          >
            9
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-eCApGN jphxbT"
        >
          <span
            class="sc-dlnjPT gbwVbp"
          >
            10
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-eCApGN kBsjdL"
        >
          <span
            class="sc-dlnjPT gbwVbp"
          >
            11
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-eCApGN jphxbT"
        >
          <span
            class="sc-dlnjPT gbwVbp"
          >
            12
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-eCApGN jphxbT"
        >
          <span
            class="sc-dlnjPT gbwVbp"
          >
            13
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-eCApGN jphxbT"
        >
          <span
            class="sc-dlnjPT gbwVbp"
          >
            <svg
              class="sc-bdnylx fIWMCQ"
              color="#FF4785"
              height="20px"
              viewBox="0 0 1024 1024"
              width="20px"
            >
              <path
                class="sc-gtssRu cdMHpp"
                color="#FF4785"
                d="M768.072 514.022c0 10.236-3.904 20.47-11.712 28.282l-344.098 344.156c-15.62 15.624-40.946 15.624-56.568 0.006-15.622-15.622-15.624-40.948-0.006-56.568l315.82-315.876-315.868-315.922c-15.618-15.624-15.618-40.952 0.004-56.568 15.624-15.62 40.95-15.618 56.57 0.006l344.144 344.204c7.81 7.81 11.714 18.044 11.714 28.28z"
              />
            </svg>
          </span>
        </button>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
